src/legal-officer/LegalOfficerContext.tsx

```diff
@@ -1,17 +1,29 @@
-import React, { useContext, useEffect, useReducer, Reducer } from 'react';
+import React, { useContext, useEffect, useReducer, Reducer, useCallback } from 'react';
 import { AxiosInstance } from 'axios';
 import { VaultTransferRequest } from '@logion/client';
 import { ProtectionRequest } from '@logion/client/dist/RecoveryClient';
 
-import { fetchProtectionRequests, } from '../common/Model';
-import { useCommonContext } from '../common/CommonContext';
+import { fetchProtectionRequests, FetchLocRequestSpecification, fetchLocRequests, } from '../common/Model';
+import { useCommonContext, LegalOfficerEndpoint } from '../common/CommonContext';
 import { LIGHT_MODE } from './Types';
-import { AxiosFactory } from '../common/api';
+import { AxiosFactory, allUp, MultiSourceHttpClient, aggregateArrays } from '../common/api';
 import { useLogionChain } from '../logion-chain';
 import { VaultApi } from '../vault/VaultApi';
+import { DataLocType, IdentityLocType, LegalOfficerCase, LocType } from "@logion/node-api/dist/Types";
+import { LocRequest } from "../common/types/ModelTypes";
+import { DateTime } from "luxon";
+import { UUID, toDecimalString } from "@logion/node-api/dist/UUID";
+import { getLegalOfficerCasesMap } from "@logion/node-api/dist/LogionLoc";
+
+const DEFAULT_NOOP = () => {};
+
+export interface RequestAndLoc {
```

Comment:
Maybe that using client's `LocData` instead of this type would have helped in simplifying the interfaces of shared visual components (they would take a `LocData` instance instead of its components)?

Reply:
Yes, maybe. But this change is not trivial. `LocData` is a unified (or merged) view of both request and loc, while `RequestAndLoc` is a composition of both. And the logic to merge is located in the SDK.